lms/djangoapps/courseware/courses.py

```diff
@@ -415,3 +419,44 @@ def get_studio_url(course, page):
     if is_studio_course and is_mongo_course:
         studio_link = get_cms_course_link(course, page)
     return studio_link
+
+
+def get_entrance_exam_score(request, course):
+    """
+    Get entrance exam score
+    """
+    exam_key = UsageKey.from_string(course.entrance_exam_id)
+    exam_descriptor = modulestore().get_item(exam_key)
+
+    def inner_get_module(descriptor):
+        """
+        Delegate to get_module_for_descriptor.
+        """
+        field_data_cache = FieldDataCache([descriptor], course.id, request.user)
+        return get_module_for_descriptor(request.user, request, descriptor, field_data_cache, course.id)
+
+    exam_module_generators = yield_dynamic_descriptor_descendents(
+        exam_descriptor,
+        inner_get_module
+    )
+    exam_modules = [module for module in exam_module_generators]
+    return calculate_entrance_exam_score(request.user, course, exam_modules)
+
+
+def get_entrance_exam_content_info(request, course):
+    """
+    Get the entrance exam content information e.g. chapter, exam passing state.
+    return exam chapter and its passing state.
+    """
+    required_content = get_required_content(course, request.user)
+    exam_chapter = None
+    is_exam_passed = True
+    if required_content:
+        # database lookup to required content pointer
+        usage_key = course.id.make_usage_key_from_deprecated_string(required_content[0])
```

Comment:
nit pick: normally when I see a [0] dereference it is because it is a current feature limitation that we can only support one item. Can you add a comment why we're just taking the first piece of required_content, e.g. what our assumptions are and how it might change in the future.